Sources/TUSKit/Files.swift

```diff
@@ -81,30 +89,79 @@ final class Files {
+            
+            if doesExist {
+                let uuidDirs = try self.contentsOfDirectory(directory: type(of: self).documentsDirectory.appendingPathComponent("TUS/background"))
+                
+                let decoder = PropertyListDecoder()
+                
+                try uuidDirs.compactMap { uuidDir in
+                    let uuidDirContents = try contentsOfDirectory(directory: uuidDir)
+                    let metaDataUrls = uuidDirContents.filter{ $0.pathExtension == "plist" }
+                    if(metaDataUrls.isEmpty) {
+                        try FileManager.default.removeItem(at: uuidDir)
+                    } else {
+                        let metaDataUrl = metaDataUrls[0]
+                        print("URL: \(metaDataUrl)")
+                        if let data = try? Data(contentsOf: metaDataUrl) {
+                            let metaData = try? decoder.decode(UploadMetadata.self, from: data)
+                            metaData?.fileDir = metaDataUrl.deletingLastPathComponent()
+                            
+                            guard let metaData = metaData else {
+                                return
+                            }
+                            
+                            // Insert into MMKV
+                            try self.encodeAndStore(metaData: metaData)
+                            
+                            // Move file to TUS/ directory
+                            try self.copyAndChunk(from: metaDataUrl, id: metaData.id, chunkSize: -1)
+                            
+                            // Add upload manifest to queue
+                            guard let uploadManifestId = metaData.context?[UPLOAD_MANIFEST_METADATA_KEY] else {
+                                return
+                            }
+                            self.addFileToUploadManifest(uploadManifestId, uuid: metaData.id)
+                            
+                            // Delete directory in TUS/background
+                            try FileManager.default.removeItem(at: uuidDir)
+                        }
+                    }
+                }
+            }
+        }
+    }
+    
     func printFileDirContents(url: URL) throws {
-        let contents = try self.contentsOfDirectory(directory: url)
+        let contents = try self.contentsOfDirectory(directory: self.storageDirectory)
         print(contents)
+        
+        let contents2 = try self.contentsOfDirectory(directory: url)
+        print(contents2)
     }
     
     /// UploadQueue cannot exist in memory because it will need to be read from the background upload side of things which won't have access to TUSClient memory
     /// So load it from file every time you need to access it as well as write it back to disk every time you update the UploadQueue
     func loadUploadQueue() throws -> UploadQueue {
-        let decoder = PropertyListDecoder()
```

Comment:
Changed all decoders from property list to JSON so that way if we need to the react-native side can easily JSON.parse items from MMKV